linera-core/src/chain_worker/state/temporary_changes.rs

```diff
@@ -172,6 +174,26 @@ where
             Box::pin(chain.execute_block(block, local_time, round.multi_leader(), None)).await?
         };
 
+        // Verify that no event values are overwritten.
+        let mut new_events = HashMap::new();
+        for event in outcome.events.iter().flatten() {
+            let event_id = event.id(chain.chain_id());
+            if let Some(old_value) = new_events.insert(event_id.clone(), &event.value) {
```

Comment:
How could this happen though? This should be a fatal error b/c it means there's problem with the code

Reply:
As long as we only have this particular system event, yes.

But we may add other operations where the user can cause a particular event to be written, or allow contracts to create events. In those cases, we should simply fail the block.

Reply:
Like I wrote many times - I think events should have index that identifies its "order" in the block. Then we'll never have this problem.

Reply:
Yes, I think you're right. We should remove the key entirely for now, and just make it a number that increments each time we write an event to a channel.

(Also a bigger change though, that I won't be able to do this week.)